src/sys/socket/mod.rs

```diff
@@ -69,23 +69,39 @@ pub enum SockType {
 
 /// Constants used in [`socket`](fn.socket.html) and [`socketpair`](fn.socketpair.html)
 /// to specify the protocol to use.
-#[repr(i32)]
 #[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
-pub enum SockProtocol {
-    /// TCP protocol ([ip(7)](http://man7.org/linux/man-pages/man7/ip.7.html))
-    Tcp = libc::IPPROTO_TCP,
-    /// UDP protocol ([ip(7)](http://man7.org/linux/man-pages/man7/ip.7.html))
-    Udp = libc::IPPROTO_UDP,
-    /// Allows applications and other KEXTs to be notified when certain kernel events occur
+pub enum SockProtocol {   
+    /// htons conversion for RAW type
+    Htons(u16),
+    /// Allows applications to configure and control a KEXT
     /// ([ref](https://developer.apple.com/library/content/documentation/Darwin/Conceptual/NKEConceptual/control/control.html))
     #[cfg(any(target_os = "ios", target_os = "macos"))]
-    KextEvent = libc::SYSPROTO_EVENT,
-    /// Allows applications to configure and control a KEXT
+    KextControl,
```

Comment:
Just out of curiosity, why did you reorder the items? :)

Reply:
Alphabetic ordering :)